src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -0,0 +1,679 @@
+      if (response.data.result?.next?.cursor === undefined) {
+        break;
+      }
+      cursor = response.data.result.next.cursor ?? undefined;
+    }
+  }
+
+  /**
+   * A list of users in reverse chronological order based on sign up.
+   */
+  public async *fetchRecentUsers({ pageSize = 100 } = {}): AsyncGenerator<
+    User,
+    void,
+    undefined
+  > {
+    let cursor: string | undefined;
+    let viewer: number | undefined;
+
+    while (true) {
+      // fetch one page of casts (with refreshed auth if necessary)
+      const response = await this.apis.user.recentUsers(
+        viewer,
+        cursor,
+        pageSize
+      );
+
+      // Extract the users array from the response (or provide an empty array as a default)
+      const users =
+        response.data.result?.users != null ? response.data.result.users : [];
+
+      // Create an async iterable for the users
+      for (const user of users) {
+        yield user;
+      }
+
+      // prep for next page
+      if (response.data.result?.next?.cursor === undefined) {
+        break;
+      }
+      cursor = response.data.result.next.cursor ?? undefined;
+    }
+  }
+
+  /**
+   * Fetch all likes by a given user.
+   */
+  public async *fetchUserCastLikes(
+    user: { fid: number },
+    { pageSize = 100 } = {}
+  ): AsyncGenerator<ReactionWithCastMeta, void, undefined> {
+    let cursor: string | undefined;
+    let viewer: number | undefined;
+
+    while (true) {
+      // fetch one page of likes
+      const response = await this.apis.user.userCastLikes(
+        user.fid,
+        viewer,
+        pageSize,
+        cursor
+      );
+
+      // Extract the likes array from the response (or provide an empty array as a default)
+      const likes =
+        response.data.result?.likes != null ? response.data.result.likes : [];
+
+      // Create an async iterable for the likes
+      for (const like of likes) {
+        yield like;
+      }
+
+      // prep for next page
+      if (response.data.result?.next?.cursor === undefined) {
+        break;
+      }
+      cursor = response.data.result.next.cursor ?? undefined;
+    }
+  }
+
+  /**
+   * Gets the specified user via their FID (if found)
+   */
+  public async lookupUserByFid(fid: number): Promise<User | undefined> {
+    try {
+      const response = await this.apis.user.user(fid);
+      return response.data.result?.user;
+    } catch (error) {
+      if (NeynarAPIClient.isApiErrorResponse(error)) {
+        if (error.response.status === 404) return undefined;
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Gets the specified user via their username (if found)
+   */
+  public async lookupUserByUsername(
+    username: string
+  ): Promise<User | undefined> {
+    let viewer: number | undefined;
+    const response = await this.apis.user.userByUsername(username, viewer, {
+      validateStatus: (status: number) => {
+        return status === 200 || status === 404;
+      },
```

Comment:
Why is this the only function where `validateStatus` is specified? Other functions are checking for 404 errors. Do we have test coverage for those other functions returning 404s?

Reply:
removed